schema/deploy/functions/handle_emission_intensity_form_change_commit.sql

```diff
@@ -0,0 +1,116 @@
+-- Deploy cif:functions/handle_emission_intensity_form_change_commit to pg
+-- requires: tables/form_change
+
+
+begin;
+
+create or replace function cif_private.handle_emission_intensity_form_change_commit(cif.form_change)
+  returns int as $$
+declare
+  reporting_requirement_record_id int;
+begin
+
+  -- If there is no change in the form data, return the form_change record and do not touch the associated table.
+  if ($1.new_form_data = '{}') then
+    return $1.form_data_record_id; -- can be null if creating with empty form data...problem?
+  end if;
+
+  if ($1.change_status = 'committed') then
+    raise exception 'Cannot commit form_change. It has already been committed.';
+  end if;
+
+  reporting_requirement_record_id := $1.form_data_record_id;
+
+  if $1.operation = 'create' then
+
+      insert into cif.reporting_requirement(
+      project_id,
+      report_type,
+      report_due_date,
+      submitted_date,
+      comments,
+      reporting_requirement_index
+    ) values (
+      (select form_data_record_id from cif.form_change pfc where form_data_table_name = 'project' and pfc.project_revision_id = $1.project_revision_id),
+      'TEIMP',
+      ($1.new_form_data->>'reportDueDate')::timestamptz,
+      ($1.new_form_data->>'submittedDate')::timestamptz,
+      ($1.new_form_data->>'comments'),
+      1
+    ) returning id into reporting_requirement_record_id;
+
+    insert into cif.emission_intensity_report(
+        reporting_requirement_id,
+        measurement_period_start_date,
+        measurement_period_end_date,
+        emission_functional_unit,
+        production_functional_unit,
+        baseline_emission_intensity,
+        target_emission_intensity,
+        post_project_emission_intensity,
+        total_lifetime_emission_reduction,
+        adjusted_emissions_intensity_performance,
+        adjusted_holdback_payment_amount,
+        date_sent_to_csnr
+    ) values (
+      reporting_requirement_record_id,
+      ($1.new_form_data->>'measurementPeriodStartDate')::timestamptz,
+      ($1.new_form_data->>'measurementPeriodEndDate')::timestamptz,
+      ($1.new_form_data->>'emissionFunctionalUnit')::varchar,
+      ($1.new_form_data->>'productionFunctionalUnit')::varchar,
+      ($1.new_form_data->>'baselineEmissionIntensity')::numeric,
+      ($1.new_form_data->>'targetEmissionIntensity')::numeric,
+      ($1.new_form_data->>'postProjectEmissionIntensity')::numeric,
+      ($1.new_form_data->>'totalLifetimeEmissionReduction')::numeric,
+      ($1.new_form_data->>'adjustedEmissionsIntensityPerformance')::numeric,
+      ($1.new_form_data->>'holdbackAmountToDate')::numeric,
+      ($1.new_form_data->>'dateSentToCsnr')::timestamptz
+    );
+
+
+    update cif.form_change set form_data_record_id = reporting_requirement_record_id where id = $1.id;
```

Comment:
unrelated question: why do we get the value of `reporting_requirement_record_id` from the form data on line 22 and then set the same value here (line 71)?

Reply:
Because the milestone form change commit does that and I copied it 😛  (I think because it gets updated line 41)